library/core/src/unit.rs

```diff
@@ -14,8 +15,8 @@ use crate::iter::FromIterator;
 /// assert!(res.is_ok());
 /// ```
 #[stable(feature = "unit_from_iter", since = "1.23.0")]
-impl FromIterator<()> for () {
-    fn from_iter<I: IntoIterator<Item = ()>>(iter: I) -> Self {
-        iter.into_iter().for_each(|()| {})
+impl<T> FromIterator<T> for () {
+    fn from_iter<A: IntoIterator<Item = T>>(iter: A) -> Self {
+        iter.into_iter().for_each(|_| {})
```

Comment:
```suggestion
        iter.into_iter().for_each(drop)
```
Maybe this should also use `.for_each(drop)` to clarify the intent better? It doesn't matter much for `()`, but for generic `T` that seems better.